pkg/arvo/neo/cod/std/src/fil/messenger.md

```diff
@@ -0,0 +1,649 @@
+# Tutorial 3: Messenger
+The last major aspect of shrubbery that we need to cover is the dependency system. This consists of the `+deps` arm and the `%rely` poke.
+
+By the end of this tutorial you’ll understand how dependencies work and how to use them. You should also start to see how you can design functionality that involves multiple shrubs interacting across multiple ships.
+
+We’ll take a look at Messenger. Messenger is one shrub, located at `/imp/messenger.hoon`, but it relies on five other shrubs to work.
+- `/imp/messenger` is the high-level interface and “service provider” that enables the user to create groupchats and 1-to-1 DMs, and invite other ships to them.
+- `/imp/message` is a `~` stub that allows us to create messages in the namespace
+- `/imp/message-pub` is a shrub that takes a poke to store messages in the namespace as its kids.
+- `/imp/message-sub` mirrors messages from a `/imp/message-pub` shrub into its own state.
+- `/imp/groupchat` creates a publisher/subscriber pair, and can invite other ships to post/subscribe to the `/imp/message-pub` shrub.
+- `/imp/dm` negotiates a two-way pub/sub relationship and mirrors state between both parties.
+
+One motivation behind this design is to split off functionality into simple, reusable shrubs. `/imp/dm` needn’t just be the DM shrub for `/messenger`, it could also be used off-the-shelf for 1-to-1 chats in shrubs like `/chess`, `/twitter`, or `/ebay`.
+
+We'll skip covering the Messenger frontend. While there are new ideas here, it's very similar to the Tasks tutorial which is the better context for more frontend. This tutorial will focus on several shrubs interoperating to form one tool. Let's look at `/imp/message-sub`, then the whole system that Messenger uses to manage chats.
+
+## /imp/message-sub
```

Comment:
Briefly touch on message-pub prior to this.

"Why does message-pub take both messages and txt?" because it might be nice to not have to manually specify the time and author sometimes, as a primitive